crates/sui-indexer/src/config.rs

```diff
@@ -208,8 +214,93 @@ pub enum Command {
 
 #[derive(Args, Default, Debug, Clone)]
 pub struct PruningOptions {
-    #[arg(long, env = "EPOCHS_TO_KEEP")]
-    pub epochs_to_keep: Option<u64>,
+    /// Path to TOML file containing configuration for retention policies.
+    #[arg(long)]
+    pub pruning_config_path: Option<PathBuf>,
+}
+
+/// Represents the default retention policy and overrides for prunable tables. When `finalize` is
+/// called, the `policies` field is updated with the default retention policy for all tables that do
+/// not have an override specified. Instantiated only if `PruningOptions` is provided on indexer
+/// start.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RetentionPolicies {
+    /// Default retention policy for all tables.
+    pub epochs_to_keep: u64,
+    /// In Rust, a mapping of all `PrunableTable` variants to their respective retention policies.
+    /// From a TOML file, a user only needs to explicitly list tables that should override the
+    /// default retention.
+    #[serde(default, rename = "overrides")]
+    pub policies: HashMap<PrunableTable, u64>,
```

Comment:
We are going to need a few default values set in the policies, while allowing then to be overridden.
For instance, we want to set the full_objects_history table to have 2-day retention by default, if not overridden.

Reply:
You meant `objects_history` table right? We can just override it in `finalize` function, if a new override value is not provided of course.

Reply:
updated to account for this